File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/KView.java
package org.kevoree.modeling.api;

import org.kevoree.modeling.api.meta.MetaClass;

public interface KView {

    KObject createByName(String metaClassName);

    KObject create(MetaClass clazz);

    void select(String query, Callback<KObject[]> cb);

    void lookup(long key, Callback<KObject> cb);

    void lookupAll(long[] keys, Callback<KObject[]> cb);

    long universe();

    long now();

    KModelFormat json();

    KModelFormat xmi();

    boolean equals(Object other);

    void setRoot(KObject elem, Callback cb);

    void getRoot(Callback<KObject> cb);

}

File: org.kevoree.modeling.microframework.typescript/src/main/ts/org/kevoree/modeling/api/KView.ts
///<reference path="meta/MetaClass.ts"/>
///<reference path="time/TimeTree.ts"/>
///<reference path="../../../../java/util/List.ts"/>
///<reference path="../../../../java/util/Set.ts"/>

interface KView {

  createFQN(metaClassName: string): KObject<any,any>;

  create(clazz: MetaClass): KObject<any,any>;

  setRoot(elem: KObject<any,any>): void;

  createJSONSerializer(): ModelSerializer;

  createJSONLoader(): ModelLoader;

  createXMISerializer(): ModelSerializer;

  createXMILoader(): ModelLoader;

  createModelCompare(): ModelCompare;

  createModelCloner(): ModelCloner<any>;

  createModelSlicer(): ModelSlicer;

  select(query: string, callback: Callback<KObject<any,any>[]>): void;

  lookup(key: number, callback: Callback<KObject<any,any>>): void;

  lookupAll(keys: number[], callback: Callback<KObject<any,any>[]>): void;

  stream(query: string, callback: Callback<KObject<any,any>>): void;

  metaClasses(): MetaClass[];

  metaClass(fqName: string): MetaClass;

  dimension(): KDimension<any,any,any>;

  now(): number;

  createProxy(clazz: MetaClass, timeTree: TimeTree, key: number): KObject<any,any>;

  listen(listener: ModelListener): void;

}

